### PR TITLE
added memory option to submission scripts

### PR DESCRIPTION
**Pull request by SalomeRonja**
My jobs have been getting stuck in the queue on Euler for longer wait times recently. I know that Benjamin experienced the same issue. At the moment, we're reserving replicas\*nmpi_per_replica\*1024MB of memory for every submitted job, which is the default for a job submitted to Euler. This results in our jobs getting submitted to the bigmem queue, even though we actually need much less memory in practice.

Example: for a production run of my 28 benzenes in water, I reserve 376'000.00 MB, the maximally used memory during the simulation is only 10'280 MB.

To reduce the amount of memory that's reserved for a job, I extended the option which already existed for steps a and b to the rest of the submission scripts. I also cleaned up and adapted the example pipeline accordingly.

### CODE REVIEW

**Comment by SalomeRonja** [comment]
The example pipeline is still running at the moment, but once it's finished without any issues, we can merge this into the eoff rebalancing branch, if you're okay with the changes @SchroederB

**Comment by SalomeRonja** [comment]
from my side, the merge is good to go :)

**Review by SchroederB — CHANGES_REQUESTED** [review]
First of all, thanks for adding the memory option!

I just found 2-3 function calls, where I think we do not pass all declared parameters, could you please check if this is true and add them quickly?  Otherwise I think this is ready to go!

File: examples/template_RE_EDS_project/a_job_opt_states.py [hunk 1]
```diff
@@ -29,7 +29,13 @@
...
-                  job_duration=job_duration, nmpi_per_replica=nmpi_per_replica,
-                  in_gromosXX_bin_dir=gromosXX_bin, in_gromosPP_bin_dir=gromosPP_bin, ene_ana_lib=ene_ana_lib)
+
+optimizeStates.do(in_simSystem=system,
+  in_imd_template_path=in_template_md_imd, 
+  out_root_dir=out_gOptStates_dir,
+  in_gromosXX_bin_dir=gromosXX_bin, 
+  in_gromosPP_bin_dir=gromosPP_bin,
+  simulation_steps = 10000,
+  ene_ana_lib = ene_ana_lib
```
[the remark this answers is not in the document]
Comment (SalomeRonja) [reply examples/template_RE_EDS_project/a_job_opt_states.py:39]: Yes, you're right... that must have gotten lost in the merge. I'll add the missing options back in! Thanks for noticing

**Review by SchroederB — APPROVED** [review]

**Pull request merged**